src/main/java/com/lorepo/icplayer/client/module/text/WCAGUtils.java

```diff
@@ -204,7 +238,23 @@ private static int getGapEndIndex(String text, int gapIndex) {
 				final int endBreakIndex = breakIndex + BREAK_TEXT.length();
 				text = text.substring(endBreakIndex);
 			}
+			if (isClosestLink) {
+				JavaScriptUtils.log("isClosestLink");
+				text = text.trim();
+				int endLinkTagIndex = text.indexOf(LINK_END);
+				String linkName = text.substring(linkIndex + LINK_START.length(), endLinkTagIndex);
+				
+				if (linkIndex > 1) {
+					result.add(TextToSpeechVoice.create(text.substring(0, linkIndex - 1), lang));                           // text before gap
+				}
+				result.add(TextToSpeechVoice.create("Link" + " " + gapNumber++));
```

Comment:
W TTSie link czytany jest jako LINK i numer gapy. Jeżeli mamy tylko linki w tekście to jest w porządzku bo liczba linków równa jest liczbie gap. Ale kiedy są gapy to TTS nie czyta ich z liczbami porządkowymi, a linki już tak, stąd jeżeli są 3 gapy a później link to on czyta jako LINK 4 - nie wiem czy tak ma być, ale może być to mylące bo link jest teoretycznie pierwszy (w tej sekcji gap) a dodatkowo pozostałe gapy nie wyliczają swojego numeru porządkowego.